gusto/equations.py

```diff
@@ -236,6 +236,8 @@ def __init__(self, field_names, domain, linearisation_map=None,
         # we should not assume this and should instead specify which variable
         # is in which space
         self.spaces = [space for space in domain.compatible_spaces]
+        if len(self.field_names) > len(self.spaces):
```

Comment:
maybe this should be a more specific check? we're assuming that just one more space is required but we don't check how much longer the list of field names is than the list of spaces... there should at least be a comment that we are assuming that any additional fields live on the DG function space